dashboard/test/controllers/api/v1/user_school_infos_controller_test.rb

```diff
@@ -0,0 +1,125 @@
+
+    assert_response :success
+    assert user_school_info.last_confirmation_date.to_datetime > original_user_school_info.to_datetime
+
+    refute_equal original_user_school_info_created_at, user_school_info[:updated_at]
+    refute_equal original_user_school_info.to_datetime, user_school_info.last_confirmation_date.to_datetime
+
+    Timecop.return
+  end
+
+  test "will redirect user to sign in" do
+    patch "/api/v1/user_school_infos/-1/update_last_confirmation_date"
+    assert_response 302
+  end
+
+  test "last confirmation date will 404 if user school info id does not exist" do
+    user_school_info = create :user_school_info
+    sign_in user_school_info.user
+    patch "/api/v1/user_school_infos/-1/update_last_confirmation_date"
+    assert_response 404
+  end
+
+  test 'update last confirmation date will 401 if given a school_info id not owned by the signed-in user' do
+    user_school_info1 = create :user_school_info
+    user_school_info2 = create :user_school_info
+    sign_in user_school_info2.user
+    patch "/api/v1/user_school_infos/#{user_school_info1.id}/update_last_confirmation_date"
+    assert_response 403
+  end
+
+  test "end_date, last_confirmation_date and last_seen_school_info_interstitial are updated" do
+    Timecop.freeze
+
+    user_school_info = create :user_school_info
+    sign_in user_school_info.user
+
+    puts "user_school_info #{user_school_info.inspect}"
+
+    Timecop.travel 1
+
+    patch "/api/v1/user_school_infos/#{user_school_info.id}/update_end_date"
+    updated_user_school_info = UserSchoolInfo.find(user_school_info.id)
+
+    assert_response :success
+
+    refute_equal user_school_info.end_date, updated_user_school_info.end_date
+    refute_equal user_school_info.last_confirmation_date, updated_user_school_info.last_confirmation_date
+    refute_equal user_school_info.user.last_seen_school_info_interstitial, updated_user_school_info.user.last_seen_school_info_interstitial
+
+    Timecop.return
+  end
+
+  test "end_date and last_seen_school_info_interstitial are not updated if given a school_info id not owned by the signed-in user." do
+    user_school_info3 = create :user_school_info
+    user_school_info4 = create :user_school_info
+    sign_in user_school_info4.user
+    patch "/api/v1/user_school_infos/#{user_school_info3.id}/update_end_date"
+
+    assert_response 403
+  end
+
+  test "school_info_id is updated" do
+    user_school_info = create :user_school_info
+    sign_in user_school_info.user
+
+    patch "/api/v1/user_school_infos/#{user_school_info.id}/update_school_info_id", params: {
+      school: {name: 'C school', city: 'Hungtinton river', state: 'Caliii'},
+      school_info: {school_type: 'private', state: 'Jersey New', school_name: 'C school', country: 'US'}
+    }
+
+    new_user_school_info = UserSchoolInfo.last
+
+    assert_response :success
+    assert UserSchoolInfo.count, 1
+    assert_not_equal new_user_school_info.id, user_school_info.id
```

Comment:
```suggestion
    refute_equal new_user_school_info.id, user_school_info.id
```

Reply:
Not blocking merge, I don't think...

Reply:
oh, I didn't realize `assert_not_equal` was actually a thing

I'm not a fan of the fact that we have so many equivalent and synonymous assert/refute methods scattered around our various test platforms.